### PR TITLE
VAULT: update state tracker now accounts for multiple epochs passing when advancing state enqueued withdrawn VRT

### PR DESCRIPTION
**Pull request by coachchucksol**
To address H-02, Ive added the ability to shift the state of VRTs enqueued to withdraw - this mimics how the operators state is updated over multiple epochs. 

Tests have also been added to cover this edgecase.

### CODE REVIEW

File: vault_program/src/close_update_state_tracker.rs [hunk 1]
```diff
@@ -42,6 +42,10 @@ pub fn process_close_vault_update_state_tracker(
     load_signer(payer, true)?;
 
     let current_ncn_epoch = slot.checked_div(config.epoch_length()).unwrap();
+    let last_updated_epoch = vault
+        .last_full_state_update_slot()
+        .checked_div(config.epoch_length())
+        .unwrap();
```
Comment (ebatsell) [inline vault_program/src/close_update_state_tracker.rs:48]: replace unwrap

Comment (coachchucksol) [reply vault_program/src/close_update_state_tracker.rs:48]: Thanks!

**Review by buffalu — CHANGES_REQUESTED** [review]
can you add test for the bug this fixes?

**Comment by coachchucksol** [comment]
The error code I've added is because we want to use checked_div. There is no way to make the epoch_length 0, so this can never happen. However, to add some clarity to it, I made a config helper function that will calculate the epoch given a slot. I have also added some - cant-be-zero checks in the integration tests for future proofing.

**Review by buffalu — APPROVED** [review]

File: integration_tests/tests/vault/crank_vault_update_state_tracker.rs [hunk 2]
```diff
@@ -635,4 +636,378 @@ mod tests {
...
+            .get_token_account(&get_associated_token_address(
+                &depositor.pubkey(),
+                &vault.supported_mint,
+            ))
+            .await
+            .unwrap();
+        assert_eq!(depositor_token_account.amount, MINT_AMOUNT);
+    }
+
+    /// Test that the vrt withdrawal process if updating with normal crank operations
```
Comment (ebatsell) [inline integration_tests/tests/vault/crank_vault_update_state_tracker.rs:778]: nit: update comment (same as above)

**Review by ebatsell — APPROVED** [review]

File: integration_tests/tests/vault/crank_vault_update_state_tracker.rs (hunk 2 above)
Comment (coachchucksol) [reply integration_tests/tests/vault/crank_vault_update_state_tracker.rs:778]: done

**Pull request merged**